### PR TITLE
[BugFix] query whose group-by and having clause depends on agg columns of aggragate table/mv can not use multiversion cache

### PR DESCRIPTION
**Pull request by satanson**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

When a aggregation query is performed on AGGREGATE KEYS OlapTable or MV whose columns carrying aggregate functions and this query satisfies one of condtions as follows:
1. group-by clause of AggregationNode depends on aggr columns.
2. having clauseof AggregationNode depends on aggr columns.
3. conjuncts of PlanNodes below AggregationNode depends on aggr columns.

Then multiversion cache mechanism can not be leveraged to merge the snapshot result from the cache and the delta rowsets from the disk into final result of a tablet. To see the UT in FE, some queries are showed.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 34 / 36 (94.44%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/DecodeNode.java|0|1|00.00%|[87]|
|:large_blue_circle:|com/starrocks/planner/FragmentNormalizer.java|20|21|95.24%|[475]|
|:large_blue_circle:|com/starrocks/planner/AggregationNode.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapScanNode.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/PlanNode.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/ProjectNode.java|2|2|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/planner/FragmentNormalizer.java [hunk 1]
```diff
@@ -469,4 +470,35 @@ public void createSimpleRangeMap(Collection<Long> selectedPartitionIds) {
         selectedRangeMap = Maps.newHashMap();
         selectedPartitionIds.stream().forEach(id -> selectedRangeMap.put(id, "[]"));
     }
+
+    public Set<SlotId> getAggColumnDependentSlotIds() {
+        return aggColumnDependentSlotIds;
+    }
+
+    public void setAggColumnDependentSlotIds(Set<SlotId> aggColumnDependentSlotIds) {
```
[the remark this answers is not in the document]
Comment (satanson) [reply fe/fe-core/src/main/java/com/starrocks/planner/FragmentNormalizer.java:478]: adopts the name slotsUseAggColumns?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 34 / 36 (94.44%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/DecodeNode.java|0|1|00.00%|[87]|
|:large_blue_circle:|com/starrocks/planner/FragmentNormalizer.java|20|21|95.24%|[475]|
|:large_blue_circle:|com/starrocks/planner/AggregationNode.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapScanNode.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/PlanNode.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/ProjectNode.java|2|2|100.00%|[]|

**Review by kangkaisen — APPROVED** [review]

**Review by ZiheLiu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by satanson** [comment]
@mergifyio backport branch-2.5

**Pull request merged**